ocamlbuild/solver.ml

```diff
@@ -40,15 +40,15 @@ let rec self depth on_the_go_orig target =
   let rules = Rule.get_rules () in
   let on_the_go = target :: on_the_go_orig in
 
-  dprintf 4 "==%a> %a" pp_repeat (depth, "==") Resource.print target;
+  if is_logging 4 then dprintf 4 "==%a> %a" pp_repeat (depth, "==") Resource.print target;
```

Comment:
Could you add a small comment here to explicit the intent that this redundant test is an optimization? (And also on the first `job_debug` test)